### PR TITLE
Added feature selection entry

### LINKED ISSUE #5074: [Concept Entry] Sklearn feature-selection
### Reason/inspiration (optional)

We would like a new entry on the `Feature Selection` concept under Sklearn. The entry should go in a new file under `docs/content/sklearn/concepts/feature-selection/feature-selection.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept (if applicable)
- An `Example` section that provides an example demonstrating the concept in use (if applicable)
- A `Codebyte Example` section that provides a codebyte example demonstrating the concept in use (if applicable)

Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by mamtawardhani**
### Description

Added a new concept entry for sklearn feature selection

### Issue Solved
Closes #5074 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [ ] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/sklearn/concepts/feature-selection/feature-selection.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  - 'Machine Learning'
+Tags:
+  - 'Machine Learning'
+  - 'Scikit-learn'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+**Feature Selection** is a critical step in machine learning pipelines that focuses on identifying the most significant features in a dataset. By removing irrelevant or redundant features, feature selection improves model performance, reduces overfitting, and decreases computation time. [Sklearn](https://www.codecademy.com/resources/docs/sklearn) offers a variety of methods for feature selection, including statistical tests, model-based selection, and iterative approaches.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/feature-selection/feature-selection.md:15]: These sentences are repetitive and slightly verbose. They can be rewritten as follows:

```suggestion
**Feature Selection** is a critical step in machine learning that helps identify a dataset's most relevant features, improving model performance, reducing overfitting, and decreasing computation time. 

[Sklearn](https://www.codecademy.com/resources/docs/sklearn) offers various methods for feature selection, including statistical tests, model-based selection, and iterative approaches.
```

File: content/sklearn/concepts/feature-selection/feature-selection.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+## Types of Feature Selection in Sklearn
+
+1. Variance Threshold
+- A simple baseline technique that removes features with variance below a predefined threshold.
+- Features with very low variance across samples typically contribute little to the predictive power of the model.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/feature-selection/feature-selection.md:21]: We can format these as nested lists, which will make it more readable, and use bold for the feature name as follows:

```suggestion
1. **Variance Threshold**
   - A simple baseline technique that removes features with variance below a predefined threshold.
   - Features with very low variance across samples typically contribute little to the predictive power of the model.
```

File: content/sklearn/concepts/feature-selection/feature-selection.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+2.  Univariate Feature Selection
+- Selects features based on univariate statistical tests.
+- Commonly used methods include `SelectKBest` and `SelectPercentile`.
+- Example tests:
+  - `f_classif`: For classification tasks, calculates the ANOVA F-value.
+  - `chi2`: For non-negative feature values in classification.
+  - `mutual_info_classif`: Captures non-linear dependencies for classification.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/feature-selection/feature-selection.md:29]: Similar changes to make it a nested list of items:

```suggestion
2.  **Univariate Feature Selection**
    - Selects features based on univariate statistical tests.
    - Commonly used methods include `SelectKBest` and `SelectPercentile`.
    - Example tests:
        - `f_classif`: For classification tasks, calculates the ANOVA F-value.
        - `chi2`: For non-negative feature values in classification.
        - `mutual_info_classif`: Captures non-linear dependencies for classification.
```

File: content/sklearn/concepts/feature-selection/feature-selection.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+3. Recursive Feature Elimination (RFE)
+- Iteratively fits a model and removes the least important features, refining the subset with each iteration.
+- Works best with models that provide feature importance, such as linear models or tree-based algorithms.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/feature-selection/feature-selection.md:33]: Similar changes to make it a nested list of items:

```suggestion
3. **Recursive Feature Elimination (RFE)**
    - Iteratively fits a model and removes the least important features, refining the subset with each iteration.
    - Works best with models that provide feature importance, such as linear models or tree-based algorithms.
```

File: content/sklearn/concepts/feature-selection/feature-selection.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+4. Sequential Feature Selection (SFS)
+- Sequentially adds or removes features to optimize a performance metric (e.g., accuracy).
+- Two approaches:
+  - **Forward Selection**: Starts with no features, adding one at a time.
+  - **Backward Elimination**: Starts with all features, removing one at a time.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/feature-selection/feature-selection.md:39]: Similar changes to make it a nested list of items:

```suggestion
4. **Sequential Feature Selection (SFS)**
    - Sequentially adds or removes features to optimize a performance metric (e.g., accuracy).
    - Two approaches:
        - **Forward Selection**: Starts with no features, adding one at a time.
        - **Backward Elimination**: Starts with all features, removing one at a time.
```

Comment (PragatiVerma18) [reply content/sklearn/concepts/feature-selection/feature-selection.md:39]: Here, the context for when to use forward or backward approaches is missing. We should add something like this -

```
Forward selection is useful when starting with a sparse model, while backward elimination is preferred for initially complex models.
```

File: content/sklearn/concepts/feature-selection/feature-selection.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+ 
+## Advantages of Feature Selection:
+- Enhances model accuracy and efficiency.
+- Reduces overfitting by removing irrelevant features.
+- Simplifies model interpretability.
+
+## Syntax
+
+Here’s an example syntax using `SelectKBest` for univariate selection:
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/feature-selection/feature-selection.md:48]: This can be rewritten for better readability as follows:

```suggestion
Below is an example of using `SelectKBest` for univariate feature selection:
```

File: content/sklearn/concepts/feature-selection/feature-selection.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+selector = SelectFromModel(model, prefit=True)
+X_train_selected = selector.transform(X_train)
+X_test_selected = selector.transform(X_test)
+
+# Selected feature indices
+selected_features = selector.get_support(indices=True)
+print("Selected Features:", selected_features)
+```
+
+The code about produces the output as follows:
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/feature-selection/feature-selection.md:91]: There's a typo here -

```suggestion
The code above produces the output as follows:
```

File: content/sklearn/concepts/feature-selection/feature-selection.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+## Syntax
+
+Here’s an example syntax using `SelectKBest` for univariate selection:
+
+```pseudo
+from sklearn.feature_selection import SelectKBest, f_classif
+
+# Applying SelectKBest
+selector = SelectKBest(score_func=f_classif, k=5)
+X_new = selector.fit_transform(X, y)
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/feature-selection/feature-selection.md:55]: In syntax, we should explain if any value is used. For ex, why is 5 used here? 

```suggestion
# Applying SelectKBest to select the top 5 features with the highest ANOVA F-values
selector = SelectKBest(score_func=f_classif, k=5)
X_new = selector.fit_transform(X, y)
```

File: content/sklearn/concepts/feature-selection/feature-selection.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+X, y = make_regression(n_samples=100, n_features=10, noise=10, random_state=42)
+
+# Splitting data into training and testing sets
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)
+
+# Fitting Lasso regression
+model = Lasso(alpha=0.1).fit(X_train, y_train)
+
+# Using SelectFromModel for feature selection
+selector = SelectFromModel(model, prefit=True)
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/feature-selection/feature-selection.md:82]: The term `prefit=True` in `SelectFromModel` might be unclear to beginners. Add a comment explaining its use:

```suggestion
# Use the pre-trained Lasso model for feature selection
selector = SelectFromModel(model, prefit=True)
```

File: content/sklearn/concepts/feature-selection/feature-selection.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+from sklearn.model_selection import train_test_split
+
+# Generating a synthetic regression dataset
+X, y = make_regression(n_samples=100, n_features=10, noise=10, random_state=42)
+
+# Splitting data into training and testing sets
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)
+
+# Fitting Lasso regression
+model = Lasso(alpha=0.1).fit(X_train, y_train)
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/feature-selection/feature-selection.md:79]: In the provided Lasso regression example, the parameter `alpha=0.1` is too small, which results in minimal regularization which might confuse learners here. 

When the regularization weakens, the Lasso model does not shrink any coefficients to zero, meaning all features are retained. This defeats the purpose of using Lasso for feature selection, as the goal is to identify and retain only the most important features while eliminating the less significant ones.

To demonstrate effective feature selection, increase the value of alpha to introduce stronger regularization. For example:

```py
# Fitting Lasso regression with stronger regularization
model = Lasso(alpha=0.5).fit(X_train, y_train)

# Using SelectFromModel for feature selection
selector = SelectFromModel(model, prefit=True)
X_train_selected = selector.transform(X_train)
X_test_selected = selector.transform(X_test)

# Selected feature indices
selected_features = selector.get_support(indices=True)
print("Selected Features:", selected_features)
```

Example Output (with `alpha=0.5`):

```shell
Selected Features: [0 2 3 7]
```

**Review by PragatiVerma18 — CHANGES_REQUESTED** [review]
Hey @mamtawardhani, thanks for adding this entry. I have some suggestions, please go through the comments and do the required changes. Let me know if you have any questions or need help. Thanks.

**Comment by mamtawardhani** [comment]
@PragatiVerma18 changes are made

**Review by PragatiVerma18 — APPROVED** [review]
